File: Frontend/src/server-models/cqrs/athletes/create-athlete.request.ts
import { AccountType } from 'src/server-models/enums/account-type.enum';
import { Gender } from 'src/server-models/enums/gender.enum';

export class CreateAthleteRequest {
    coachId: string;
    username: string;
    email: string;
    firstname: string;
    lastname: string;
    accountType: AccountType;
    password: string;
    confirmPassword: string;
    gender: Gender;
}

File: Backend/Core/Backend.Application.Business/Business/Athletes/CreateAthlete/CreateAthleteRequest.cs
using System;
using Backend.Domain.Entities;
using Backend.Domain.Enum;
using MediatR;

namespace Backend.Application.Business.Business.Athletes.CreateAthlete
{
    public class CreateAthleteRequest : IRequest<Athlete>
    {
        public Guid CoachId { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public string FullName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime CreatedOn => DateTime.UtcNow;
        public DateTime LastModified => CreatedOn;
        public bool Active => true;
        public AccountType AccountType => AccountType.Athlete;
        public Gender Gender { get; set; }
    }
}
